watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -737,16 +924,41 @@ def LCOW(b):
         doc="Levelized cost of water with respect to volumetric permeate flow"
     )
     def LCOW_wo_revenue(b):
-        return (
-            b.total_capital_cost * b.zo_costing.capital_recovery_factor
-            + b.total_operating_cost
-        ) / (
-            pyunits.convert(
-                b.permeate.properties[0].flow_vol,
-                to_units=pyunits.m**3 / pyunits.year,
+        if hasattr(m.fs, "dewaterer"):
+            return (
+                b.total_capital_cost * b.zo_costing.capital_recovery_factor
+                + b.total_operating_cost
+            ) / (
+                pyunits.convert(
+                    b.permeate.properties[0].flow_vol
+                    + b.centrate.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / pyunits.year,
+                )
+                * b.zo_costing.utilization_factor
+            )
+        elif hasattr(m.fs, "gac"):
+            return (
+                b.total_capital_cost * b.zo_costing.capital_recovery_factor
+                + b.total_operating_cost
+            ) / (
+                pyunits.convert(
+                    b.permeate.properties[0].flow_vol
+                    + b.treated.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / pyunits.year,
+                )
+                * b.zo_costing.utilization_factor
+            )
+        else:
+            return (
+                b.total_capital_cost * b.zo_costing.capital_recovery_factor
+                + b.total_operating_cost
+            ) / (
+                pyunits.convert(
+                    b.permeate.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / pyunits.year,
+                )
+                * b.zo_costing.utilization_factor
```

Comment:
I think this logic can be eliminated with a proper Product block that captures all relevant product water streams in any flowsheet configuration. Another item for a subsequent PR.